src/perform_predict/predict.py

```diff
@@ -97,23 +97,38 @@ def _create_input_space(input_config: Dict[str, list[Dict[str, Any]]],
     output_file_df = constants.PRED_INPUT_SPACE_FILE
 
     path = utils.write_df_to_csv(
-        df=input_space_df, 
-        output_path=output_path, 
+        df=input_space_df,
+        output_path=output_path,
         output_file=output_file_df)
     logger.info(f"Wrote input feature space to {path}")
     return path, input_space_df
 
 
-def _get_missing_values(original_data: pd.DataFrame, var_name: str):
-
-    values = range(original_data[var_name].min()+1, original_data[var_name].max())
+def _get_data_range_missing_values(original_data: pd.DataFrame, var_name: str) -> list[str]:
+    values = range(original_data[var_name].min() + 1, original_data[var_name].max())
     return [val for val in values if val not in original_data[var_name].tolist()]
 
 
+def _get_highest_ranked_estimator(estimator_path: str, target_variable: str) -> str:
+    ranking_file = os.path.join(estimator_path, constants.AM_RANKINGS_FILE)
+    if not os.path.exists(ranking_file):
+        logger.error("Failed to locate model ranking file")
+        raise Exception
+    ranking_df = pd.read_csv(ranking_file)
+    ranking_df_var = ranking_df.loc[ranking_df[constants.AM_COL_TARGET] == target_variable]
+    ranking_df_var['combined_rank'] = ranking_df_var[constants.AM_COL_RANK_MAPE] + \
```

Comment:
I am not sure about having a combined rank across two measures of prediction
error together with R^2 which speaks more to correlation. Best combined rank of
MAPE and NRMSE seems less problematic to me. Maybe we should consider not
including R^2 in this decision. What do you think?

Reply:
Good point. Removed R^2 from combined ranking.